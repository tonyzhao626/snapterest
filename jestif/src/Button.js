import styled, {css} from 'styled-components';


const Button = styled.button`
  padding: ${props => props.padding ? props.padding : '10px 20px'};
  background-color: green;
  border: none;
  color: white;
  transition: transform ${props => props.duration ? props.duration : 0}ms;
  transform-style: preserve-3d;
  ${props => props.rotate && css`
    transform: rotate${props.vertical ? 'X' : 'Y'}(180deg);
`};

& :hover {
  background-color: red;
  width: 400px;
}
`;
export default Button;
