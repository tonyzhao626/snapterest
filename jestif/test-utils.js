import React from 'react'
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import defaultTheme from './src/theme';


export const matches = (children, theme = defaultTheme) => expect(
  renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ).toJSON()
).toMatchSnapshot()