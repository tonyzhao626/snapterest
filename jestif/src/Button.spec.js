import React from 'react';
import { matches } from '../test-utils';
import Button from './Button';

describe('Button', () => {
  it('matches snapshot', () => {
    matches(<Button>Submit</Button>);
  });
});