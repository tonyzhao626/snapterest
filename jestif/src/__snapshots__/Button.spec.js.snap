// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<button
  className="sc-bdVaJa jOkEfV"
>
  Submit
</button>
`;
